### PR TITLE
Add translateOptions as props in DeleteWithConfirmButton

### PR DESCRIPTION
**Pull request by vllorca**
This is implementing a feature to create a custom message when deleting an element

# Why 

The goal is to be able to create a custom message during the confirmation of the deletion.
In our application, pictures are displayed and identified by both id and a identifier. The custom message use the id for the deletion message and in our case identifier is the one that should be use because it's more accurate.

# Solution 

This PR introduce `translateOptions` as a prop in `DeleteWithConfirmButton`, so `id` and `name` used for the custom message can be easily changed.

Example of usage :

```
import * as React from 'react';
import { Toolbar } from '../form';

translateOptions = {
    id: 'custom identifier',
    name: 'custom name'
}

const EditToolbar = props => (
    <Toolbar {...props}>
        <DeleteWithConfirmButton translateOptions={translateOptions} />
    </Toolbar>
);
```

### CODE REVIEW

**Comment by fzaninotto** [comment]
I think the code sample in your PR description is wrong.

File: docs/Buttons.md [hunk 1]
```diff
@@ -298,7 +298,20 @@ const ListActions = () => (
 
 ## Record Buttons
 
-### `<DeleteButton>`
```
Comment (fzaninotto) [inline docs/Buttons.md:301]: please don't delete this, it's a reminder ;)

File: docs/Buttons.md [hunk 2]
```diff
@@ -298,7 +298,20 @@ const ListActions = () => (
...
+
+Delete the selected element. To be used inside a <Toolbar> conponent.
+
+```jsx
+import * as React from 'react';
+import { Toolbar, SimpleForm } from '../form';
+
+const EditToolbar = props => (
+    <Toolbar {...props}>
+        <DeleteWithConfirmButton translateOptions={translateOptions} />
```
Comment (fzaninotto) [inline docs/Buttons.md:311]: This isn't explanatory. What is the format of translateOptions? Why would anyone use it?

File: packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.spec.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { render, waitFor, fireEvent } from '@testing-library/react';
+import { render, screen, waitFor, fireEvent } from '@testing-library/react';
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/button/DeleteWithConfirmButton.spec.tsx:1]: linter warning

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/Buttons.md (hunk 1 above)
Comment (vllorca) [reply docs/Buttons.md:301]: I put it back

File: docs/Buttons.md (hunk 2 above)
Comment (vllorca) [reply docs/Buttons.md:320]: I added some elements to be more explicit

File: docs/Buttons.md [hunk 4]
```diff
@@ -299,6 +299,35 @@ const ListActions = () => (
...
+}
+
+const EditToolbar = props => (
+    <Toolbar {...props}>
+        <DeleteWithConfirmButton translateOptions={translateOptions} />
+    </Toolbar>
+);
+```
+
+| Prop         | Required | Type            | Default            | Description                         |
```
Comment (fzaninotto) [inline docs/Buttons.md:325]: Without an introduction, this table is misleading - it looks like the description of the `DeleteWithConfirmButton` props

File: docs/Buttons.md [hunk 5]
```diff
@@ -299,6 +299,35 @@ const ListActions = () => (
 ## Record Buttons
 
 ### `<DeleteButton>`
+
+### `<DeleteWithConfirmButton>`
+
+Delete the selected element. To be used inside a <Toolbar> conponent.
+Overriding translateOptions prop allows to make custom message for the deletion.
```
Comment (fzaninotto) [inline docs/Buttons.md:306]:
```suggestion

Overriding `translateOptions` prop allows to customize the notification message confirming the deletion.
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/Buttons.md (hunk 4 above)
Comment (vllorca) [reply docs/Buttons.md:326]: I was inspired by previous tables in the documentation, in order to describe the changes applied by my PR (concerning the two props).
What should this table describe ?

**Comment by fzaninotto** [comment]
This PR was automatically closed by GitHub, I'm reopening it. Sorry for the noise.

File: docs/Buttons.md [hunk 6]
```diff
@@ -299,6 +299,37 @@ const ListActions = () => (
 ## Record Buttons
 
 ### `<DeleteButton>`
+
+### `<DeleteWithConfirmButton>`
+
+Delete the selected element. To be used inside a <Toolbar> conponent.
```
Comment (djhi) [inline docs/Buttons.md:305]:
```suggestion
Delete the current record. To be used inside a <Toolbar> conponent.
```

File: docs/Buttons.md [hunk 7]
```diff
@@ -299,6 +299,37 @@ const ListActions = () => (
...
+
+Overriding `translateOptions` name and id props allows to customize the notification message confirming the deletion.
```
Comment (djhi) [inline docs/Buttons.md:307]:
```suggestion
You can customize the notification message confirming the deletion by providing a `translateOptions` prop. It accepts an object with a `name` and `id` property:
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Comment by WiXSL** [comment]
Was this a valid PR?

**Comment by fzaninotto** [comment]
Yep, reopening.

**Comment by WiXSL** [comment]
Implemented by #6241

**Pull request closed without merging**